Apsitvarkom.Api/Controllers/PollutedLocationController.cs

```diff
@@ -131,4 +134,31 @@ public async Task<ActionResult<PollutedLocationResponse>> Create(PollutedLocatio
 
         return CreatedAtAction(nameof(GetById), new { response.Id }, response);
     }
-}
\ No newline at end of file
+
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(List<string>))]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HttpPut("Update")]
+    public async Task<ActionResult<PollutedLocationResponse>> Update(PollutedLocationUpdateRequest pollutedLocationUpdateRequest)
+    {
+        var validatorResponse = await _pollutedLocationUpdateValidator.ValidateAsync(pollutedLocationUpdateRequest);
+        if (!validatorResponse.IsValid) return BadRequest(validatorResponse.Errors.Select(e => e.ErrorMessage).ToList());
+
+        var existingLocation = await _repository.GetByPropertyAsync(x => x.Id == pollutedLocationUpdateRequest.Id);
+        if (existingLocation is null) return StatusCode(StatusCodes.Status500InternalServerError);
+       
+        var mappedLocation = _mapper.Map<PollutedLocationUpdateRequest, PollutedLocation>(pollutedLocationUpdateRequest, existingLocation);
```

Comment:
this is wrong. if we send all the properties, it works as expected, but if for example we would like to only update the severity, radius, progress and notes would become the default values of those types (string? -> null, int -> 0), instead of keeping the previous values.
We should remove this mapping and instead put only the non-null variables to the recently extracted PollutedLocation instance.

Reply:
should I check each values separetely and update only those which were changed ?

Reply:
No, I don't think that it is necessary. First of all, even if the data is the same, the re-assignment with the same value is not an expensive operation, which doesn't make any difference to the end result. Secondly, update request variables can be null (meaning they're optional) - this would allow the user to not specify the value at all if they wanted to keep the value the same.